dask/dataframe/core.py

```diff
@@ -3609,6 +3609,12 @@ class DataFrame(_Frame):
     _token_prefix = "dataframe-"
     _accessors = set()
 
+    def __init__(self, dsk, name, meta, divisions):
+        super().__init__(dsk, name, meta, divisions)
+        self.dask.layers[name].type = type(self)
+        self.dask.layers[name].divisions = divisions
```

Comment:
I think `divisions` is the dataframe analogue to `chunks` in dask arrays - please correct me if I'm wrong here.